src/gui/editors/PianoRoll.cpp

```diff
@@ -3598,9 +3598,9 @@ void PianoRoll::paintEvent(QPaintEvent * pe )
 	}
 	else
 	{
-		QFont f = p.font();
-		f.setBold( true );
-		p.setFont(pointSize(f, 14));
+		QFont f = font();
```

Comment:
I think we should still be getting `p.font()` here?

Reply:
The idea is to reset the painter's font to the font of the widget and then to make some adjustments as needed. This will for example reset the font size to the default application/widget font size. The idea is to write the message in this context in the "normal" font and not scaled in any way.

Reply:
Yes, it inherits it when the `QPainter´ is constructed. However, in the case of this file (and also the other IIRC) the painter's font is already manipulated before in [line 3339](https://github.com/LMMS/lmms/blob/0c787a908ddb0d2d5898bcdd9d1f5a7ee9559087/src/gui/editors/PianoRoll.cpp#L3339) of the file.

Getting a "fresh" font from the widget ensures that we do not have to keep track of what all was done before. If I fetch it from the painter here then the font will not use it's default size and the text will be drawn too small. By doing it like this it's drawn in the default font size of the widget, in this case the application font size which should be nicely legible for the user.

Reply:
Alright, as long as we've tested that this doesn't affect any drawing output then good with me.

Reply:
Yes, it should be safe. The calls are done in `else´ blocks and are the last calls to `drawText`.

Reply:
Unrelated but would it make sense to remove the function call and use `setPixelSize` directly in this area?

Reply:
Do you mean in this block or in line 3340? In this block the goal is to render in the default point size of the application and not in pixel size.

Currently the calls to `adjustedToPixelSize` serve as nice "markers" where pixel sizes are used. I think it makes sense to remove them and replace them with direct calls to `setPixelSize` if it is clear that the implementation cannot/should not be changed to a point sized implementation anyway. This might be the case in line 3340.

Reply:
Yeah i meant to say about line 3340, mb.

Reply:
Adjusted with commit faec019455c.